backend/handler/options.go

```diff
@@ -55,5 +55,69 @@ func (s *Server) PostOption(e echo.Context, params PostOptionParams) error {
 }
 
 func (s *Server) PutOption(e echo.Context, optionID OptionId, params PutOptionParams) error {
-	return nil
+	user, err := s.repo.GetOrCreateUser(*params.XForwardedUser)
+
+	if err != nil {
+		e.Logger().Errorf("failed to get or create user: %v", err)
+
+		return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
+	}
+
+	if !user.IsStaff {
+		return echo.NewHTTPError(http.StatusForbidden, "Forbidden")
+	}
+
+	var req PutOptionJSONRequestBody
+
+	if err := e.Bind(&req); err != nil {
+		e.Logger().Errorf("failed to bind request body: %v", err)
+
+		return echo.NewHTTPError(http.StatusBadRequest, "Bad request")
+	}
+
+	var updatedOption model.Option
```

Comment:
https://github.com/traP-jp/rucQ/pull/162#discussion_r1947390127 と同じ理由で、一旦既存の`Option`を持ってきてほしいです。今は`Option`単体を取得する関数がないので、それを追加する必要もありそう :eyes:
```suggestion
	updatedOption, err := s.repo.GetOption(uint(optionID))
	
	if err != nil {
		if errors.Is(err, model.ErrNotFound) {
			return echo.NewHTTPError(http.StatusNotFound, "Not found")
		}
		e.Logger().Errorf("failed to get option: %v", err)
		return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
	}
```